server/src/main/java/org/opensearch/index/mapper/WildcardFieldMapper.java

```diff
@@ -0,0 +1,825 @@
+                assert indexAnalyzers != null;
+                normalizer = indexAnalyzers.getNormalizer(normalizerName);
+            }
+
+            return new WildcardFieldMapper(
+                name,
+                new WildcardFieldType(context.path().pathAsText(name), normalizer, this),
+                multiFieldsBuilder.build(this, context),
+                copyTo.build(),
+                this
+            );
+        }
+
+    }
+
+    public static final String CONTENT_TYPE = "wildcard";
+    public static final TypeParser PARSER = new TypeParser((n, c) -> new WildcardFieldMapper.Builder(n, c.getIndexAnalyzers()));
+
+    protected WildcardFieldMapper(
+        String simpleName,
+        MappedFieldType mappedFieldType,
+        MultiFields multiFields,
+        CopyTo copyTo,
+        Builder builder
+    ) {
+        super(simpleName, mappedFieldType, multiFields, copyTo);
+        this.nullValue = builder.nullValue.getValue();
+        this.ignoreAbove = builder.ignoreAbove.getValue();
+        this.normalizerName = builder.normalizer.getValue();
+        this.hasDocValues = builder.hasDocValues.getValue();
+        this.indexAnalyzers = builder.indexAnalyzers;
+    }
+
+    public int ignoreAbove() {
+        return ignoreAbove;
+    }
+
+    private static final FieldType FIELD_TYPE = new FieldType();
+    static {
+        FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
+        FIELD_TYPE.setTokenized(true);
+        FIELD_TYPE.setOmitNorms(true);
+        FIELD_TYPE.setStored(false);
+        FIELD_TYPE.freeze();
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context) throws IOException {
+        String value;
+        if (context.externalValueSet()) {
+            value = context.externalValue().toString();
+        } else {
+            XContentParser parser = context.parser();
+            if (parser.currentToken() == XContentParser.Token.VALUE_NULL) {
+                value = nullValue;
+            } else {
+                value = parser.textOrNull();
+            }
+        }
+
+        if (value == null || value.length() > ignoreAbove) {
+            return;
+        }
+
+        NamedAnalyzer normalizer = fieldType().normalizer();
+        if (normalizer != null) {
+            value = normalizeValue(normalizer, name(), value);
+        }
+
+        // convert to utf8 only once before feeding postings/dv/stored fields
+        final BytesRef binaryValue = new BytesRef(value);
+        Tokenizer tokenizer = new WildcardFieldTokenizer();
+        tokenizer.setReader(new StringReader(value));
+        context.doc().add(new Field(fieldType().name(), tokenizer, FIELD_TYPE));
+        if (fieldType().hasDocValues()) {
+            context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
+        } else {
+            if (fieldType().hasDocValues() == false) {
+                createFieldNamesField(context);
+            }
+        }
+    }
+
+    /**
```

Comment:
My thinking was that it doesn't hurt to include them, since there are not very many of them. 

While e.g `*a*` or `*e*` will probably behave like a match-all (or field-exists) query anyway, a search for `*z*` or `*q*` might get a bit of a speedup.

My intuition is that bigrams will be somewhat useful, since `*ab*` will probably match a much smaller proportion of docs.

Of course, at query time, as long as a candidate fragment is 3 or more characters, we only use trigrams.

It might be worth benchmarking to see how much smaller the index gets if we exclude unigrams or both unigrams and bigrams.

Reply:
thanks for your reasoning. It would certainly be an interesting exercise to see size differences because of uni and bigrams.